unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Controllers/BIWPublishController.cs

```diff
@@ -23,6 +24,9 @@ public override void Init()
             HUDController.i.builderInWorldMainHud.OnPublishAction += StartPublishFlow;
             HUDController.i.builderInWorldMainHud.OnConfirmPublishAction += StartPublishScene;
         }
+
+        builderInWorldBridge = InitialSceneReferences.i.builderInWorldBridge;
```

Comment:
Suggestion: maybe it would be fine to pass the `InitialSceneReferences` as injected reference in the `BIWController.Init()` function and not have to always depend of the singleton instance. In this way we would be following the dependency injection philosofy and would ease the testing using mocks (in this case, for example we could have a mocked `InitialSceneReferences` with a mocked `builderInWorldBridge` and we wouldn't have to depend of a "real" instance of `InitialSceneReferences`).

The same for the rest of `Init()` functions of each `BIWController`. I think it could be a good practice to apply.

Anyway let me know waht do you think? I may be missing something that I'm not taking into account 👍

Reply:
The problem with that is that `InitialSceneReference ` can destroy his current instance (imagine that we reload the initial scene, or load an additional scene that changes the scenes references).  Thats been said, it is probably that this won't happen. 

`InitialSceneReference `  is a special class that only references the monobehaviours of the initial scene, I think this is the only case where the Singleton pattern is OK. 

I think that create a mocked `InitialSceneReference ` instance won't be a good choice since it will need to instantiate gameobjects (This is the current solution for reference monobehaviour in scene) and it will create more overheat to the test increasing their duration.

I can do that without problem but that depends on the future implementation of `InitialSceneReference`

@BrianAmadori  what do you think?

Reply:
Yeah, I think it is a good momento to open this discussion and decide what approach to follow in these cases. For me, the less dependencies a unit test has on a Unity scene, the better. This would force us to create the needed game objects in runtime but I think the non-dependency of anything continues having more benefits.

Reply:
I think we can keep the singleton but in this case inject the dependency. This would help us to get rid of the scene references in the future as I stated in the other comment. I think that using a singleton is a last resort tool, we should try to inject dependencies if it makes sense.